src/mps/mps.jl

```diff
@@ -257,7 +257,7 @@ function sample(m::MPS)
       projn = ITensor(s)
       projn[s[n]] = 1.0
       An = A*projn
-      pn = scalar(dag(An)*An)
+      pn = scalar(dag(An)*An) |> real_if_close
```

Comment:
Did you see a problem occur here if you just use `real(scalar(dag(An)*An))`? That result should definitely be real (up to floating point precision). It seems like checking for `imag(pn) < tol` should be more like a debug check.

Reply:
No, that'd work; it just makes me nervous to give up the check. If some value fails to make sense I want the program to choke early, rather than silently continuing and (maybe) giving nonsensical results. Especially for something like this, where the symptom's going to be that some average just ... doesn't quite look right; tracking that down to find some subtle bug like accidentally producing an imaginary part sounds miserable.

On the other hand, I can't imagine a case where `scalar(dag(An)*An))` _isn't_ real (up to numerical precision), provided `dag` is working as it should.

On the gripping hand, my imagination is limited...